pinot-segment-local/src/main/java/org/apache/pinot/segment/local/upsert/ConcurrentMapPartitionUpsertMetadataManager.java

```diff
@@ -69,6 +69,9 @@ protected void addOrReplaceSegment(ImmutableSegmentImpl segment, ThreadSafeMutab
     String segmentName = segment.getSegmentName();
     segment.enableUpsert(this, validDocIds, queryableDocIds);
 
+    if (_partialUpsertHandler != null) {
+      recordInfoIterator = resolveComparisonTies(recordInfoIterator, _hashFunction);
+    }
```

Comment:
We can move this to Base class too. But i see we will have to make add this logic in multiple methods. Ideally we should have a base method implementation doing this (something like `doAddOrReplaceSegment` and then calling `addOrReplaceSegment` here.
Maybe we can take the refactoring in a follow-up PR.

Reply:
@Jackie-Jiang : this small refactor suggestion looks good to me. Though I'd probably change the name a bit. Since by usual naming conventions, a method like `addOrReplaceSegment` would be the one calling `doAddOrReplaceSegment`.

lmk your thoughts.